### PR TITLE
KK-123 | Add edit child modal

### PR DESCRIPTION
**Pull request by midori0507**
- Extra: clear token right when http request error to ensure prevent rendering loop ?
- Refactor birthdate form data population.

### CODE REVIEW

File: src/domain/child/ChildUtils.ts [hunk 1]
```diff
@@ -48,3 +50,31 @@ export const getTranslatedRelationshipOptions = (
 export const getSupportedChildData = (child: Child) => {
   return omit(child, 'homeCity');
 };
+
+/**
+ * Convert from birthdate in Date string format
+ * to object of day month year, mostly to furfill form data structure
```
Comment (badrange) [inline src/domain/child/ChildUtils.ts]: furfill -> fulfill

File: src/domain/profile/children/child/ProfileChildDetail.tsx [hunk 2]
```diff
@@ -77,6 +81,17 @@ const ProfileChildDetail: React.FunctionComponent = () => {
                     alt={t('profile.child.detail.edit.icon.alt')}
                   />
                 </Button>
+                {isOpen && (
+                  <ProfileChildDetailEditModal
+                    isOpen={isOpen}
+                    setIsOpen={setIsOpen}
+                    //TODO: Remove homeCity if backend future support
+                    edittingChild={{ ...child, ...{ homeCity: '' } }}
```
Comment (badrange) [inline src/domain/profile/children/child/ProfileChildDetail.tsx]: edittingChild -> editingChild
and I wonder if childBeingEdited is even better?

Comment (badrange) [reply src/domain/profile/children/child/ProfileChildDetail.tsx]: or childToEdit

File: src/domain/child/__tests__/ChildUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    });
+
+    test('should return child birthdate in format for form by default if given wrong format date', () => {
+      const childForm = getChildFormModalBirthdate('2020-02-31');
+
+      expect(childForm).toEqual(defaultChildFormBirthdate);
+    });
+
+    test('should give a nice form format birthdate data when given valid birthdate string', () => {
+      const childForm = getChildFormModalBirthdate('2020-02-27');
```
Comment (badrange) [inline src/domain/child/__tests__/ChildUtils.test.ts]: Would be great to test leap year too! 2020-02-28 exists, but not 2021-02-28

File: src/domain/profile/children/child/modal/ProfileChildDetailEditModal.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import { Child } from '../../../../child/types/ChildTypes';
+import ChildFormModal from '../../../../child/modal/ChildFormModal';
+import { getChildFormModalValues } from '../../../../child/ChildUtils';
+
+const ProfileChildDetailEditModal: React.FunctionComponent<{
+  isOpen: boolean;
+  setIsOpen: (value: boolean) => void;
+  editChild: (payload: Child) => void;
+  edittingChild: Child;
```
Comment (badrange) [inline src/domain/profile/children/child/modal/ProfileChildDetailEditModal.tsx]: edittingChild -> editingChild
and I wonder if childBeingEdited is even better?

File: src/domain/child/__tests__/ChildUtils.test.ts (hunk 3 above)
Comment (midori0507) [reply src/domain/child/__tests__/ChildUtils.test.ts]: > image
> 
> I found a bug! The city is not prefilled. We should probably prefill it with Helsinki (en/fi)/Helsingfors(sv) if the variable is empty.

Added default homeCity for child edit form

**Pull request merged**